src/realm/data_type.hpp

```diff
@@ -19,8 +19,21 @@
 #ifndef REALM_DATA_TYPE_HPP
 #define REALM_DATA_TYPE_HPP
 
+#include "realm/string_data.hpp"
+#include "realm/binary_data.hpp"
+#include "realm/olddatetime.hpp"
+
```

Comment:
None of the includes are actually needed here. StringData and BinaryData can just be forward-declared and the OldDateTime typedef is redundant.